src/RobotInterface/YarpImplementation/src/YarpImplementation.cpp

```diff
@@ -0,0 +1,554 @@
+                      << std::endl;
+            return false;
+        }
+
+        // store the polydriver
+        this->robotDevice = robotDevice;
+
+        // obtain the interfaces
+        if (!robotDevice->view(encodersInterface) || encodersInterface == nullptr)
+        {
+            std::cerr << errorPrefix << "Cannot load the IEncodersTimed interface." << std::endl;
+            return false;
+        }
+
+        if (!robotDevice->view(positionInterface) || positionInterface == nullptr)
+        {
+            std::cerr << errorPrefix << "Cannot load the IPositionControl interface." << std::endl;
+            return false;
+        }
+
+        if (!robotDevice->view(positionDirectInterface) || positionDirectInterface == nullptr)
+        {
+            std::cerr << errorPrefix << "Cannot load the IPositionDirect interface." << std::endl;
+            return false;
+        }
+
+        if (!robotDevice->view(velocityInterface) || velocityInterface == nullptr)
+        {
+            std::cerr << errorPrefix << "Cannot load the IVelocityInterface interface."
+                      << std::endl;
+            return false;
+        }
+
+        if (!robotDevice->view(torqueInterface) || torqueInterface == nullptr)
+        {
+            std::cerr << errorPrefix << "Cannot load the ITorqueInterface interface." << std::endl;
+            return false;
+        }
+
+        if (!robotDevice->view(controlModeInterface) || controlModeInterface == nullptr)
+        {
+            std::cerr << errorPrefix << "Cannot load the IControlMode interface." << std::endl;
+            return false;
+        }
+
+        if (!robotDevice->view(axisInfoInterface) || axisInfoInterface == nullptr)
+        {
+            std::cerr << errorPrefix << "Cannot load the IAxisInfo interface." << std::endl;
+            return false;
+        }
+
+        // get the number of degree of freedom
+        int dofs = 0;
+        if (!encodersInterface->getAxes(&dofs))
+        {
+            std::cerr << errorPrefix << "Cannot get the actuated DoFs." << std::endl;
+            return false;
+        }
+        this->actuatedDOFs = dofs;
+
+        // resize vector
+        this->positionFeedback.resize(this->actuatedDOFs);
+        this->controlModes.resize(this->actuatedDOFs);
+        this->controlModesYarp.resize(this->actuatedDOFs);
+        this->axesName.resize(this->actuatedDOFs);
+
+
+        // populate the axesName vector
+        for (int i = 0; i < this->actuatedDOFs; i++)
+        {
+            this->axisInfoInterface->getAxisName(i, this->axesName[i]);
+        }
+
+        return this->getControlModes();
+    }
+
+    std::pair<int, double> getWorstPositionDirectError(Eigen::Ref<const Eigen::VectorXd> desiredJointValues,
+                                                       Eigen::Ref<const Eigen::VectorXd> jointPositions) const
+    {
+        // clear the std::pair
+        std::pair<int, double> worstError{0, 0.0};
```

Comment:
I suppose the pair describes the joint index and the error? Could you make the comment more descriptive?